File: esercizi_python/compito_python_3/Giocatore.py
from Prenotazioni import Prenotazioni


class Giocatore:
    def __init__(self, cod, nome, eta, categoria):
        self.cod = cod
        self.nome = nome
        self.eta = eta
        self.categoria = categoria
        self.prenotazione = []

    def addPrenotazione(self, c, orainiz):
        self.prenotazione.append(Prenotazioni(c, orainiz))
    
    def ciaone(self):
        x= ""
        for p in self.prenotazione:
            x = x+ p.toString()
        return  x
    
    def volte(self, b):
        a = 0
        for p in self.prenotazione:
            if b == p.c.cod:
                a = p.c.costorario+a
        return a




    def toString(self):
        return str(self.cod)+"\t"+self.nome+"\t"+str(self.eta)+"\t"+str(self.categoria)+"\t["+self.ciaone()+"]"

File: esercizi_java/NetBeansProjects/Circolo_Sportivo/src/circolo_sportivo/Giocatore.java
package circolo_sportivo;
import java.util.List;
import java.util.LinkedList;
/**
 *
 * @author tobiasacchetto
 */
public class Giocatore {
    protected int cod,eta,categoria;
    protected String nome;
    protected List<Prenotazioni> prenotazione;
    public Giocatore(int cod, String nome, int eta, int categoria){
        this.cod=cod;
        this.categoria=categoria;
        this.nome=nome;
        this.eta=eta;
        this.prenotazione = new LinkedList<Prenotazioni>();
    }
    
    public void addPrenotazione(Campo c, int orainiz){
        prenotazione.add(new Prenotazioni(c,orainiz));
    }
    
    public float volte(int b){
        float a=0;
        for(Prenotazioni p: prenotazione){
            if(b==p.c.cod){
                a=p.c.costorario+a;
            }
        }
        return a;
    }
    
    @Override
    public String toString(){
        return cod+"\t"+nome+"\t"+eta+"\t"+categoria+"\t"+prenotazione.toString()+"";
    }
}
